apps/builder/app/builder/features/settings-panel/code-editor.tsx

```diff
@@ -0,0 +1,383 @@
+import { useCallback, useEffect, useState } from "react";
```

Comment:
is this code editor supposed to be any code editing or specifically expressions editor?

Reply:
Any code editing + completion of variables. Variables can be for js and css with custom properties.

Reply:
I just wonder how to distinguish this from a generic code editor we will have for react components at some point. And also this is gonna be needed in style panel too so settings-panel is not the right place soon enough

Reply:
I'd keep it here before it become suitable for css. Would be reimplemented anyway to support more stuff like highlighting.